lib/node_modules/@stdlib/blas/ext/base/gcusumkbn2/test/test.ndarray.js

```diff
@@ -126,6 +126,93 @@ tape( 'the function calculates the cumulative sum', function test( t ) {
 	t.end();
 });
 
+tape( 'the function calculates the cumulative sum (accessors)', function test( t ) {
+	var expected;
+	var x;
+	var y;
+	var i;
+
+	x = [ 1.0, 2.0, 3.0, 4.0, 5.0 ];
+	y = [ 0.0, 0.0, 0.0, 0.0, 0.0 ];
+
+	gcusumkbn2( x.length, 0.0, toAccessorArray( x ), 1, 0, toAccessorArray( y ), 1, 0 );
+	expected = [
+		1.0,
+		3.0,
+		6.0,
+		10.0,
+		15.0
+	];
+	t.deepEqual( y, expected, 'returns expected value' );
+
+	x = [ 1.0, 2.0, 3.0, 4.0, 5.0 ];
+	y = [ 0.0, 0.0, 0.0, 0.0, 0.0 ];
+
+	gcusumkbn2( x.length, 10.0, toAccessorArray( x ), 1, 0, toAccessorArray( y ), 1, 0 );
+	expected = [
+		11.0,
+		13.0,
+		16.0,
+		20.0,
+		25.0
+	];
+	t.deepEqual( y, expected, 'returns expected value' );
+
+	x = [ NaN, NaN ];
+	y = [ 0.0, 0.0 ];
+	gcusumkbn2( x.length, 0.0, toAccessorArray( x ), 1, 0, toAccessorArray( y ), 1, 0 );
+
+	for ( i = 0; i < y.length; i++ ) {
+		t.strictEqual( isnan( y[ i ] ), true, 'returns expected value. i: ' + i );
+	}
+
+	x = [ 1.0, NaN, 3.0, NaN ];
+	y = [ 0.0, 0.0, 0.0, 0.0 ];
+	gcusumkbn2( x.length, 0.0, toAccessorArray( x ), 1, 0, toAccessorArray( y ), 1, 0 );
+
+	expected = [
+		1.0,
+		NaN,
+		NaN,
+		NaN
+	];
+	for ( i = 0; i < y.length; i++ ) {
```

Comment:
As an aside, the manual loop is fine with me, but, for reference, we also recently added [`isSameArray`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/assert/is-same-array), which abstracts array such manual iteration.